File: today/triangle.ts
/**
 * https://leetcode.com/problems/triangle/description/
 */
export { minimumTotal };

function minimumTotal(triangle: number[][]): number {
  const memo: Record<string, number> = {};

  function dfs(row: number, col: number): number {
    let key = `${row},${col}`;
    if (key in memo) return memo[key];
    if (row === triangle.length - 1) return triangle[row][col];

    const result1 = triangle[row][col] + dfs(row + 1, col);
    const result2 = triangle[row][col] + dfs(row + 1, col + 1);

    return (memo[key] = Math.min(result1, result2));
  }

  return dfs(0, 0);
}

File: returned/python/today10/triangle.py
from typing import List

class Solution:
    def minimumTotal(self, triangle: List[List[int]]) -> int:
        eval = triangle[len(triangle)-1]
        for row in range(len(triangle)-2, -1, -1):
            next = []
            for col in range(0, len(triangle[row])):
                 smallest = triangle[row][col] + min(eval[col], eval[col+1])
                 next.append(smallest)
            eval = next
        return eval[0]